mlir/lib/Dialect/OpenMP/IR/OpenMPDialect.cpp

```diff
@@ -1748,11 +1754,27 @@ LogicalResult WsloopOp::verify() {
   if (!isWrapper())
     return emitOpError() << "must be a loop wrapper";
 
+  auto wrapper =
+      llvm::dyn_cast_if_present<LoopWrapperInterface>((*this)->getParentOp());
+  bool isCompositeWrapper = wrapper && wrapper.isWrapper() &&
+                            (!llvm::isa<ParallelOp>(wrapper) ||
+                             llvm::isa<DistributeOp>(wrapper->getParentOp()));
```

Comment:
Same comment for `SimdOp::verify()`.
```suggestion
                             llvm::isa_and_present<DistributeOp>(wrapper->getParentOp()));
```